mantidproject/mantid
# Save Environment Algorithm

**Description of work.**
Added algorithm `SaveSampleEnvironmentAndShape` to save out a binary .stl file containing the mesh of the entire shape of the environment and sample as a single mesh.
**To test:**
use .stl files in `ExternalData/Testing/Data/UnitTest/` along with `LoadSampleShape` and `LoadSampleEnvironment` to construct a sample (it must at least have a sample shape) then run `SaveSampleEnvironmentAndShape`, the .stl file created should be loadable in mantid using `LoadSampleShape` and should be able to be loaded in https://www.viewstl.com/. 

Attempting to save a non mesh based sample, e.g. one created using `CreateSampleShape` should fail.

The new tests should pass.

Fixes #25722

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
